### PR TITLE
Current items merges to dev

### PR DESCRIPTION
**Pull request by simple-user**

### CODE REVIEW

File: net/DescriptionViewController.h [hunk 1]
```diff
@@ -13,13 +13,9 @@
...
 @interface DescriptionViewController : UIViewController
 
-// @property (strong, nonatomic) Issue *currentIssue;
-@property (strong, nonatomic) User *currentUser;
-
 @property (weak, nonatomic) IBOutlet UILabel *titleLabel;
 @property (weak, nonatomic) IBOutlet UILabel *descriptionLabel;
-@property (strong,nonatomic) UIImage *image;
-@property (nonatomic, getter=isActualImageView)BOOL actualImageView;
+//@property (strong,nonatomic) UIImage *image;
```
Comment (dscherb) [inline net/DescriptionViewController.h]: Please remove commented code

File: net/DescriptionViewController.m [hunk 2]
```diff
@@ -68,51 +66,39 @@ - (void) viewWillAppear:(BOOL)animated
...
-        }
+        NSLog(@"if(cItems.issueImage==nil) in description");
+        self.issueImageView.image = nil;
+    }
+     else if( ![self.issueImageView.image isEqual:cItems.issueImage])
+    {
+        NSLog(@"view will appear: ![self.issueImageView.image isEqual:cItems.issueImage], set uotlet.");
+        self.issueImageView.image = cItems.issueImage;
     }
-//    [self.dataSorce requestImageWithName:self.currentIssue.attachments andViewControllerHandler:^(UIImage *image) {
```
Comment (dscherb) [inline net/DescriptionViewController.m:85]: Commented code

File: net/DescriptionViewController.m [hunk 3]
```diff
@@ -68,51 +66,39 @@ - (void) viewWillAppear:(BOOL)animated
...
 {
-    _image = image;
-
-    if(self.issueImageView !=nil)
+    CurrentItems *cItems = [CurrentItems sharedItems];
+    if (![self.issueImageView.image isEqual:cItems.issueImage])
     {
-        self.issueImageView.image = image;
-        self.actualImageView = YES;
+        NSLog(@"in description issue did load: ![self.issueImageView.image isEqual:cItems.issueImage], setOutlet");
```
Comment (dscherb) [inline net/DescriptionViewController.m:87]: Really wired log string

File: net/LogInViewController.m [hunk 4]
```diff
@@ -234,22 +235,4 @@ - (IBAction)logInButton:(UIButton *)sender
 
     }];
 }
-
-
-//#pragma mark - Navigation
```
Comment (dscherb) [inline net/LogInViewController.m:239]: Commented code

File: net/MapViewController.m [hunk 5]
```diff
@@ -103,7 +105,7 @@ -(void)viewWillAppear:(BOOL)animated
         self.currentUser = [CurrentItems sharedItems].user;
     }
     
-    if(self.currentMarker!=nil)
+    if([self isMarkerSelected])
```
Comment (dscherb) [inline net/MapViewController.m]: Why not if (self.isMarkerSelected) ?

File: net/MapViewController.m [hunk 6]
```diff
@@ -144,35 +146,12 @@ -(void)setCurrentUser:(User *) user
     }
 }
 
-#pragma mark - Navigation
-//-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
```
Comment (dscherb) [inline net/MapViewController.m:148]: Commented code

File: net/MapViewController.m [hunk 7]
```diff
@@ -287,34 +266,19 @@ -(BOOL)mapView:(GMSMapView *)mapView didTapMarker:(GMSMarker *)marker
         [self.view layoutIfNeeded];
         
     }];
-    self.currentMarker = marker;
+    self.isMarkerSelected = YES;
     
-    DescriptionViewController *descriptionVC = nil;
```
Comment (dscherb) [inline net/MapViewController.m:292]: Commented code

File: net/MapViewController.m [hunk 8]
```diff
@@ -395,23 +359,14 @@ -(BOOL)tabBarController:(UITabBarController *)tabBarController shouldSelectViewC
     if ([viewController isKindOfClass:[UINavigationController class]] && [viewController.restorationIdentifier isEqualToString:@"description"]){
         UINavigationController *destController = (UINavigationController *)viewController;
         DescriptionViewController *descriptionVC = (DescriptionViewController *)destController.topViewController;
-//        descriptionVC.currentIssue = self.currentMarker.userData;
```
Comment (dscherb) [inline net/MapViewController.m:398]: Commented code

File: net/MapViewController.m [hunk 9]
```diff
@@ -395,23 +359,14 @@ -(BOOL)tabBarController:(UITabBarController *)tabBarController shouldSelectViewC
     if ([viewController isKindOfClass:[UINavigationController class]] && [viewController.restorationIdentifier isEqualToString:@"description"]){
```
Comment (dscherb) [inline net/MapViewController.m]: Are we steel need this method?

File: net/MapViewController.m [hunk 10]
```diff
@@ -478,13 +433,6 @@ -(void)animateTabsSwitching:(UIViewController *)viewController
     
     UIView *fromView = self.tabBarController.selectedViewController.view;
     UIView *toView = [viewController view];
-    //UIView *toView = [self.tabBarController.viewControllers[controllerIndex] view];
```
Comment (dscherb) [inline net/MapViewController.m:481]: Are we steel need this method?

File: net/model/CurrentItems.h [hunk 11]
```diff
@@ -6,11 +6,22 @@
 //  Copyright © 2016 Admin. All rights reserved.
 //
 
+#import <Foundation/Foundation.h>
+@protocol UserImageDelegate <NSObject>
+
+@optional
```
Comment (dscherb) [inline net/model/CurrentItems.h]: Why optional?

File: net/model/CurrentItems.h [hunk 12]
```diff
@@ -6,11 +6,22 @@
...
+-(void)userImageDidLoad;
+@end
+
+@protocol IssueImageDelegate <NSObject>
+
+@optional
```
Comment (dscherb) [inline net/model/CurrentItems.h]: Same here

File: net/model/NetworkDataSorce.m [hunk 13]
```diff
@@ -139,12 +142,13 @@ -(void)requestSingUpWithUser:(User*)user
                  andDataSorceHandler:^(NSData *data, NSError *error) {
                      if(data.length >0 && error == nil)
                      {
-                         NSDictionary *userDic = [NSJSONSerialization JSONObjectWithData:data
+                         NSMutableDictionary *userDic = [[NSJSONSerialization JSONObjectWithData:data
```
Comment (dscherb) [inline net/model/NetworkDataSorce.m:145]: Do you really need to mutableCopy this dictionary?

File: net/DescriptionViewController.m (hunk 3 above)
Comment (simple-user) [inline net/DescriptionViewController.m:87]: It's just logging! I'll remove it

File: net/MapViewController.m (hunk 5 above)
Comment (simple-user) [inline net/MapViewController.m]: It's just a style. But,to tell the truth, it doesn't really matter for me, which of two to use.
Besides as usual, when I use dot notation, I'll use it explicitly 
if (self.isMarkerSelected == YES)

File: net/MapViewController.m (hunk 10 above)
Comment (simple-user) [inline net/MapViewController.m:481]: Unfortunatelly, for now, we don't have generall opinion about using (or not) animation with changing viewcontrollers with tabbar.

File: net/MapViewController.m (hunk 9 above)
Comment (simple-user) [inline net/MapViewController.m]: I haven't already implemented notification to update map from description. So for now I still using Map delegate :(

File: net/model/CurrentItems.h (hunk 11 above)
Comment (simple-user) [inline net/model/CurrentItems.h]: Roger! :) This is really looks strange.

File: net/model/NetworkDataSorce.m (hunk 13 above)
Comment (simple-user) [inline net/model/NetworkDataSorce.m:145]: um.... When I get json from server (user or issue), and it hasn't image (avatar or attach), I'll get [NSNull null]. At few next lines I save info into NSUserDefaults and app crashes cos [NSNull null] is not a property list. So I check it, and modify Dictionary.

**Pull request closed without merging**